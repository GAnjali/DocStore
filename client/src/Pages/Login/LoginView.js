import React from "react";
import { Link } from "react-router-dom";
import {
  DOCSTORE,
  EMAIL_PLACEHOLDER,
  LOGIN,
  NEED_AN_ACCOUNT,
  PASSWORD_PLACEHOLDER,
  SIGNUP_URL
} from "../../AppConstants";

const LoginView = props => {
  return (
    <div className="center">
      <div className="card">
        <h1>{DOCSTORE}</h1>
        <form>
          <input
            className="form-item"
            placeholder={EMAIL_PLACEHOLDER}
            name="email"
            type="text"
            onChange={props.handleInputChange}
          />
          <input
            className="form-item"
            placeholder={PASSWORD_PLACEHOLDER}
            name="password"
            type="password"
            onChange={props.handleInputChange}
          />
          {props.hasError && <h6 className={"error"}>{props.error}</h6>}
          <input
            className="form-submit"
            value={LOGIN}
            type="submit"
            onClick={props.handleLogin}
          />
        </form>
        <div className={"signupLink"}>
          {NEED_AN_ACCOUNT}
          <Link
            onClick={() => {
              props.history.push(SIGNUP_URL);
            }}
          >
            Signup
          </Link>
        </div>
      </div>
    </div>
  );
};

export default LoginView;