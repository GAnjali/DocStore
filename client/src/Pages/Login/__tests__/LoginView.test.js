import React from "react";
import {shallow} from "enzyme";
import LoginView from "../components/LoginView";

describe("LoginView tests", () => {
    const props = {
        handleChange: jest.fn(),
        handleFormSubmit: jest.fn(),
        hasError: false,
        error: null
    };
    it("Should render without crashing", () => {
        shallow(<LoginView props/>);
    });
    it('should match the LoginView snapshot', function () {
        const loginViewComponent = shallow(<LoginView props/>);
        expect(loginViewComponent).toMatchSnapshot();
    });
});
