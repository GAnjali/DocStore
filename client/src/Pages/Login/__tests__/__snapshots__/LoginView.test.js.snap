// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView tests should match the LoginView snapshot 1`] = `ShallowWrapper {}`;
